scheduler/filter/whitelist.go

```diff
@@ -0,0 +1,71 @@
+package filter
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/swarm/cluster"
+	"github.com/docker/swarm/scheduler/node"
+)
+
+// WhitelistFilter selects only nodes that are defined in a whitelist by the user
+type WhitelistFilter struct {
+}
+
+// Name returns the name of the filter
+func (f *WhitelistFilter) Name() string {
+	return "whitelist"
+}
+
+// Filter is exported
+func (f *WhitelistFilter) Filter(config *cluster.ContainerConfig, nodes []*node.Node, soft bool) ([]*node.Node, error) {
+	whitelists, err := parseExprs(config.Whitelists())
+	if err != nil {
+		return nil, err
+	}
+
+	for _, whitelist := range whitelists {
```

Comment:
At a second thought, this way is more consistent with other filters like `constraints`. Let's keep it as it is.